src/App.js

```diff
@@ -1,21 +1,30 @@
-function App() {
+import { useDispatch, useSelector } from "react-redux";
+
+import userFetch from "./store/user/asyncActions";
+
+const App = () => {
+  const currentUser = useSelector((state) => state.user);
```

Comment:
you should store this selector in a `store/user/selectors.js` file
this selector should return the direct information about user (to use it as currentUser.firstName but not currentUser.data.firstName)

and currentUser.status should be replaced with a separated selector for that
it's better to have a lot of different selectors